src/SDL_net.c

```diff
@@ -0,0 +1,1662 @@
+#include "SDL3_net/SDL_net.h"
+
+#ifdef __WINDOWS__
+#define WIN32_LEAN_AND_MEAN 1
+#define _WIN32_WINNT 0x0600  /* we need APIs that didn't arrive until Windows Vista. */
```

Comment:
I see this warning on ci:
```
D:/a/SDL_net/SDL_net/src/SDL_net.c:5:9: warning: '_WIN32_WINNT' macro redefined [-Wmacro-redefined]
#define _WIN32_WINNT 0x0600  /* we need APIs that didn't arrive until Windows Vista. */
        ^
D:/a/_temp/msys64/clang64/include/_mingw.h:232:9: note: previous definition is here
#define _WIN32_WINNT 0x603
```
Do we need to define this macro in cmake?

https://github.com/libsdl-org/SDL_net/actions/runs/6328625286/job/17187212344#step:8:19

Reply:
> I see this warning on ci:

Oh, whoops, we should probably enable -Werror now.  :)

Maybe we should see if it's defined and less than 0x600, and #undef it if so, first.

Reply:
> > I see this warning on ci:
> 
> Oh, whoops, we should probably enable -Werror now. :)
> 
> Maybe we should see if it's defined and less than 0x600, and #undef it if so, first.

I don't know whether it'd have an effect if you do that: That we hit this warning means that we somehow included win32 api headers already. Defining from cmake may be an easier solution.

Reply:
Yeah, SDL_net.h includes SDL.h, which I assume includes it _somewhere_.

I'd rather avoid doing this in CMake unless absolutely necessary; I could delete the change I made and just make sure it's `#define`d before we including _anything_ in SDL_net.c ...?